packages/core/src/context/hotkeys/hotkeysProvider.tsx

```diff
@@ -122,9 +123,14 @@ export const HotkeysProvider = ({ children, dialogProps, renderDialog, value }:
         />
     );
 
+    const reducedState = React.useMemo<HotkeysContextState>(
+        () => ({ hasProvider: state.hasProvider }),
+        [state.hasProvider],
+    );
+
     // if we are working with an existing context, we don't need to generate our own dialog
     return (
-        <HotkeysContext.Provider value={[state, dispatch]}>
+        <HotkeysContext.Provider value={[reducedState, dispatch]}>
```

Comment:
(totally aware this is a draft) But I think we actually want to memo this reference itself too. eg.

```ts
const memoValue = React.useMemo(() => [reducedState, dispatch], [])
```

Also have seen this show up in some perf-profiles, but haven't had the time to follow-up. Without a memo, each time this provider renders, it also re-renders every single consuming child (even if state/dispatch) haven't changed.